src/Core/Addon/Theme/ThemeManager.php

```diff
@@ -136,29 +136,32 @@ public function enable($name)
             return false;
         }
 
-        $theme = $this->getInstanceByName($name);
-        if (!$this->theme_checker->isValid($theme)) {
+        $theme = $this->themeRepository->getInstanceByName($name);
+        if (!$this->themeValidator->isValid($theme)) {
             return false;
         }
 
         $this->disable($this->shop->theme_name);
 
         $this->doCreateCustomHooks($theme->get('global_settings.hooks.custom_hooks', []))
-                ->doApplyConfiguration($theme->get('global_settings.configuration', []))
-                ->doDisableModules($theme->get('global_settings.modules.to_disable', []))
-                ->doEnableModules($theme->get('global_settings.modules.to_enable', []))
-                ->doHookModules($theme->get('global_settings.hooks.modules_to_hook'));
+            ->doApplyConfiguration($theme->get('global_settings.configuration', []))
+            ->doDisableModules($theme->get('global_settings.modules.to_disable', []))
+            ->doEnableModules($theme->get('global_settings.modules.to_enable', []))
+            ->doHookModules($theme->get('global_settings.hooks.modules_to_hook', []))
+        ;
 
         $theme->onEnable();
 
         $this->shop->theme_name = $theme->getName();
         $this->shop->update();
 
+        $this->saveTheme($theme);
```

Comment:
Nitpicking here, but now that we have a `ThemeRepository`, shouldn't the `saveTheme` method be on it and not `ThemeManager`?